R/utils.R

```diff
@@ -259,85 +259,560 @@ rowcol_to_dim <- function(row, col) {
+#' * `col_names` `wb_dims()` assumes that if `x` has column names, then trying to find the `dims`.
+#'
+#'  You can use `unname(x)` to give better input
+#'
+#'
+#' `wb_dims()` tries to support most possible cases with `row_names = TRUE` and `col_names = FALSE`,
+#' but it works best if `x` has named dimensions (`data.frame`, `matrix`), and those parameters are not specified.
+#'  data with column names, and without row names. as the code is more clean.
+#'
+#' In the `add_data()` / `add_font()` example, if writing the data with row names
+#'
+#'
+#' ```r
+#' dims_row_names <- wb_dims(x = mtcars, row_names = TRUE, cols = 0)
+#' # add data to an object with row names
+#' wb <- wb_workbook()
+#' wb$add_worksheet("test")
+#' full_mtcars_dims <-
+#' wb$add_data(x = mtcars, dims = wb_dims(x = mtcars, row_names = TRUE), row_names = TRUE)
+#' # Style row names of an object (many options)
+#' # The programmatic way to access row names only with `x` is
+#' dims_row_names <- wb_dims(x = mtcars, row_names = TRUE, col_names = FALSE, cols = 0, from_col = 0)
+#' # In this case, it's much better to use a simpler alternative without using `x`
+#' dims_row_names <- wb_dims(cols = "A", from_row = 2)
+#' dims_row_names <- wb_dims(2:33, 1) # or dims <- "A2:A33"
+#' dims_row_names <- "A2:A33" # or simply "A2"
+#' wb$add_font(dims = dims_row_names, bold = TRUE)
+#' # the following would work too, but `wb_dims()` may be longer to write, but easier to read after, as
+#' # it can make it clear which object is affected
+#' wb$add_font(dims = dims_row_names, bold = TRUE)
+#'
+#' ```
+#'
+#' @param ... construct dims arguments, from rows/cols vectors or objects that
+#'   can be coerced to data frame
+#' @return A `dims` string
+#' @export
 #' @examples
-#' # either vectors
+#' wb_dims(x = mtcars, row_names = TRUE, col_names = FALSE, cols = 0)
+#' # Provide coordinates
+#' wb_dims()
+#' wb_dims(1, 4)
+#' wb_dims(rows = 1, cols = 4)
+#' wb_dims(from_row = 4)
+#' wb_dims(from_col = 2)
+#' wb_dims(1:4, 6:9, from_row = 5)
+#' # Provide vectors
+#' wb_dims(1:10, c("A", "B", "C"))
 #' wb_dims(rows = 1:10, cols = 1:10)
+#' # provide `from_col` / `from_row`
+#' wb_dims(rows = 1:10, cols = c("A", "B", "C"), from_row = 2)
+#' wb_dims(rows = 1:10, cols = 1:10, from_col = 2)
 #' # or objects
-#' wb_dims(mtcars)
-#' @export
+#'
+#' wb_dims(x = mtcars)
+#' # column names of an object (with the special `rows = 0`)
+#' wb_dims(x = mtcars, rows = 0)
+#' # usually, it's better
+#' # dims of all the data of mtcars.
+#' wb_dims(x = mtcars, col_names = FALSE)
+#'
+#' # dims of the column names of an object
+#' wb_dims(x = mtcars, rows = 0, col_names = TRUE)
+#'
+#' ## add formatting to column names with the help of `wb_dims()` ====
+#' wb <- wb_workbook()
+#' wb$add_worksheet("test")
+#' wb$add_data(x = mtcars, dims = wb_dims(x = mtcars))
```

Comment:
I agree that this may not be the right thing
Imo `wb_dims()` with `x` works better and makes more sense with column names

# work with mtcars full
wb_dims(x = mtcars, col_names = TRUE)
"A1:K33"
# work with mtcars data
wb_dims(x = mtcars)
 "A2:K33"
# work with mtcars without column nameds
wb_dims(x = unname(mtcars), col_names = FALSE)



But do you agree with this ?
```r
wb_dims(x = mtcars, cols = "cyl")
#>  "B2:B33"
```

Because when applying conditional styling, you usually don't want to touch the column name.

And use this code this
```r
wb_dims(x = mtcars, cols = "cyl", col_names = TRUE)
#>  "B1:B33"
```

Reply:
Yes, I agree with the latter part. If a column is selected we can default to ignore the column name row unless it is explicitly requested. Though the default should be to include the column name with data frames.
I just saw that you proposed defaulting to `col_names = FALSE` in your initial proposal, sorry for not seeing it in the first place and now being picky.